Source/WTF/wtf/FlipBytes.h

```diff
@@ -40,76 +41,15 @@ inline bool needToFlipBytesIfLittleEndian(bool littleEndian)
 #endif
 }
 
-inline uint16_t flipBytes(uint16_t value)
-{
-    return ((value & 0x00ff) << 8)
-        | ((value & 0xff00) >> 8);
-}
-
-inline uint32_t flipBytes(uint32_t value)
-{
-    return ((value & 0x000000ff) << 24)
-        | ((value & 0x0000ff00) << 8)
-        | ((value & 0x00ff0000) >> 8)
-        | ((value & 0xff000000) >> 24);
-}
-
-inline uint64_t flipBytes(uint64_t value)
-{
-    return ((value & 0x00000000000000ffull) << 56)
-        | ((value & 0x000000000000ff00ull) << 40)
-        | ((value & 0x0000000000ff0000ull) << 24)
-        | ((value & 0x00000000ff000000ull) << 8)
-        | ((value & 0x000000ff00000000ull) >> 8)
-        | ((value & 0x0000ff0000000000ull) >> 24)
-        | ((value & 0x00ff000000000000ull) >> 40)
-        | ((value & 0xff00000000000000ull) >> 56);
-}
-
 template<typename T>
 inline T flipBytes(T value)
 {
-    if (sizeof(value) == 1)
-        return value;
-    if (sizeof(value) == 2) {
-        union {
-            T original;
-            uint16_t word;
-        } u;
-        u.original = value;
-        u.word = flipBytes(u.word);
-        return u.original;
-    }
-    if (sizeof(value) == 4) {
-        union {
-            T original;
-            uint32_t word;
-        } u;
-        u.original = value;
-        u.word = flipBytes(u.word);
-        return u.original;
-    }
-    if (sizeof(value) == 8) {
-        union {
-            T original;
-            uint64_t word;
-        } u;
-        u.original = value;
-        u.word = flipBytes(u.word);
-        return u.original;
-    }
-    if (sizeof(value) == 16) {
-        union {
-            T original;
-            uint64_t words[2];
-        } u, v;
-        v.original = value;
-        u.words[0] = flipBytes(v.words[1]);
-        u.words[1] = flipBytes(v.words[0]);
-        return u.original;
-    }
-    RELEASE_ASSERT_NOT_REACHED();
-    return T();
+    if constexpr (std::is_integral_v<T>)
+        return std::byteswap(value);
+
+    auto byteRepresentation = std::bit_cast<std::array<std::byte, sizeof(T)>>(value);
+    std::ranges::reverse(byteRepresentation);
+    return std::bit_cast<T>(byteRepresentation);
```

Comment:
Does this end up getting used, or is it dead code? It really seems like the original intention here is much more the `std::byteswap` version.

Reply:
Maybe we can use a `uint128_t` in IsoBox though if that's the only case (I did not check).